### PR TITLE
Home Tabs

### PR DESCRIPTION
**Pull request by Marcosnr97**
### Summary
Added a layout of tabs to home using viewpager2 and tablayout.

### Screenshots
Peek 2021-04-19 10-45

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/home/ViewPagerAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.Lifecycle
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import ar.com.wolox.android.example.ui.home.news.NewsFragment
+import ar.com.wolox.android.example.ui.home.userProfile.UserProfileFragment
+
+class ViewPagerAdapter(fragment: FragmentManager, lifecycle: Lifecycle) : FragmentStateAdapter(fragment, lifecycle) {
+
+    override fun getItemCount(): Int = 2
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/ViewPagerAdapter.kt]: 👁️

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.kt [hunk 2]
```diff
@@ -40,11 +40,20 @@ class LoginPresenter @Inject constructor(
...
         if (!loginError) {
-            view?.showLoginLoading()
+            view?.toggleProgressBarPresence(true)
             networkRequest(userRepository.loginUser(user)) {
-                onResponseSuccessful { _ -> view?.logInUser() }
-                onResponseFailed { _, _ -> view?.showLoginError() }
-                onCallFailure { view?.showConnectionError() }
+                onResponseSuccessful {
+                    view?.logInUser()
+                    view?.toggleProgressBarPresence(false)
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.kt]: move `view?.toggleProgressBarPresence(false)` to te bottom of the `networkRequest` method.

File: app/src/main/java/ar/com/wolox/android/example/ui/viewpager/fragment/ViewPagerFragment.kt [hunk 3]
```diff
@@ -27,7 +27,8 @@ class ViewPagerFragment private constructor() : WolmoFragment<FragmentViewpagerB
     override fun handleArguments(arguments: Bundle?) = arguments?.containsKey(FAVOURITE_COLOR_KEY)
 
     override fun init() {
-        binding.viewPager.adapter = SimpleFragmentPagerAdapter(childFragmentManager).apply {
+        binding.viewPager
+        .adapter = SimpleFragmentPagerAdapter(childFragmentManager).apply {
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/viewpager/fragment/ViewPagerFragment.kt]: this indentation seems a little weird and you are not using this file 👀

File: app/src/main/java/ar/com/wolox/android/example/ui/home/fragment/HomeFragment.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import com.google.android.material.tabs.TabLayoutMediator
+import javax.inject.Inject
+
+class HomeFragment private constructor() : WolmoFragment<FragmentHomeBinding, HomePresenter>(), HomeView {
+
+    private lateinit var homeAdapter: ViewPagerAdapter
+    private lateinit var pager: ViewPager2
+
+    @Inject
+    internal lateinit var newsFragment: NewsFragment
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/fragment/HomeFragment.kt:20]: I think you don't need to make this and the UserProfileFragment `internal`

File: app/src/main/java/ar/com/wolox/android/example/ui/home/fragment/HomeFragment.kt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        with(binding) {
+            homeAdapter = ViewPagerAdapter(requireActivity().supportFragmentManager, requireActivity().lifecycle)
+            pager = viewPager
+            pager.adapter = homeAdapter
+            tabLayout.tabMode = TabLayout.MODE_FIXED
+            TabLayoutMediator(tabLayout, pager) { tab, position ->
+                when (position) {
+                    0 -> {
+                        tab.text = getString(R.string.news_tab)
+                        tab.setIcon(R.drawable.ic_news_list)
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/fragment/HomeFragment.kt]: You can use a Kotlin function like `with(tab)`

**Review by ffasitella — CHANGES_REQUESTED** [review]
Just a few comments, good job! 😄

File: app/src/main/java/ar/com/wolox/android/example/ui/home/ViewPagerAdapter.kt (hunk 1 above)
Comment (Marcosnr97) [reply app/src/main/java/ar/com/wolox/android/example/ui/home/ViewPagerAdapter.kt]: oops

File: app/src/main/java/ar/com/wolox/android/example/ui/viewpager/fragment/ViewPagerFragment.kt (hunk 3 above)
Comment (Marcosnr97) [reply app/src/main/java/ar/com/wolox/android/example/ui/viewpager/fragment/ViewPagerFragment.kt]: probably a victim of a bunch of viewpager testing 😅

File: app/src/main/java/ar/com/wolox/android/example/ui/home/fragment/HomeFragment.kt (hunk 4 above)
Comment (Marcosnr97) [reply app/src/main/java/ar/com/wolox/android/example/ui/home/fragment/HomeFragment.kt:20]: In the boostrap example, ViewPagerFragment, both randomFragment and requestFragment are `internal` . For consistency i left both of my fragments internal.

**Review by ffasitella — APPROVED** [review]
Nice job! 🚀

**Review by jcruz-wolox — APPROVED** [review]

**Pull request merged**